components/sup/src/events.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2019 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Main interface for a stream of events the Supervisor can send out
+//! in the course of its operations.
+//!
+//! Currently, the Supervisor is able to send events to a [NATS
+//! Streaming][1] server. The `init_stream` function must be called
+//! before sending events to initialize the publishing thread in the
+//! background. Thereafter, you can pass "event" structs to the
+//! `event` function, which will publish the event to the stream.
+//!
+//! All events are published under the "habitat" subject.
+//!
+//! [1]:https://github.com/nats-io/nats-streaming-server
+
+mod nats;
```

Comment:
I dunno, I like the fact that all the implementation-specific stuff is off in the `nats` file, while all the "internal API" of the event system is in the `events` module.

Reply:
i kinda like the nats stuff in its own file too... makes it really obvious what we'd need to change were we to swap implementations. i'm a rust noob, tho, and def don't understand idiomatic use.

Reply:
If we foresee a *different* implementation other than `nats` as highly likely, I can get behind that, but otherwise it seems like premature generalization which has me hopping around more files to understand the code. When there's only one implementation, I think the distinction between external and internal interfaces are just the addition or omission of `pub`, and that's pretty idiomatic.

Reply:
ah, ok... thanks much!

Reply:
I've merged the nats module into the event module.